service/policy/db/subject_mappings.go

```diff
@@ -324,9 +324,17 @@ func (c PolicyDBClient) CreateSubjectConditionSet(ctx context.Context, s *subjec
 	if err != nil {
 		return nil, err
 	}
-	if err = r.Scan(&id); err != nil {
+	if err = r.Scan(&id, &metadataJSON); err != nil {
 		return nil, db.WrapIfKnownInvalidQueryErr(err)
 	}
+
+	if metadataJSON != nil {
+		if err = protojson.Unmarshal(metadataJSON, m); err != nil {
+			slog.Error("could not unmarshal metadata", slog.String("error", err.Error()))
+			return nil, err
+		}
+	}
```

Comment:
Along the lines of the other enhancements, would it make sense to abstract this as well?

```go
// the function could do the condition validation
if err := unmarshalMetadata(metadataJSON, m); err != nil
  // the function could slog the error on error
  return nil, err
}
```

Reply:
done